py-polars/polars/utils/convert.py

```diff
@@ -184,26 +180,26 @@ def _to_python_datetime(
 ) -> datetime:
     """Convert polars int64 timestamp to Python datetime."""
     if not time_zone:
-        if time_unit == "us":
+        if time_unit == "ns":
+            return EPOCH + timedelta(microseconds=value // 1_000)
+        elif time_unit == "us":
             return EPOCH + timedelta(microseconds=value)
-        elif time_unit == "ns":
-            return EPOCH + timedelta(microseconds=value // 1000)
         elif time_unit == "ms":
             return EPOCH + timedelta(milliseconds=value)
         else:
             raise ValueError(
-                f"`time_unit` must be one of {{'ns','us','ms'}}, got {time_unit!r}"
+                f"`time_unit` must be one of {{'ms', 'us', 'ns'}}, got {time_unit!r}"
```

Comment:
Any reason to revert this order? If so, it should also be reverted in `py-polars/src/conversion.rs`.

Reply:
No reason.
What I had in mind is to standarize the ValueErrors related to this. Compare with the ValueError from py-polars/polars/datatypes/classes.py:
```
raise ValueError(
    "invalid `time_unit`"
    f"\n\nExpected one of {{'ns','us','ms'}}, got {self.time_unit!r}."
)
```